cirq/ops/common_channels.py

```diff
@@ -184,6 +195,700 @@ def depolarize(p: float) -> DepolarizingChannel:
             the Pauli gates is applied independently with probability p / 3.
 
     Raises:
-        ValueError if p is not a valid probability.
+        ValueError: if p is not a valid probability.
     """
     return DepolarizingChannel(p)
+
+
+class GeneralizedAmplitudeDampingChannel(raw_types.Gate):
+    """Dampen qubit amplitudes through non ideal dissipation.
+
+    This channel models the effect of energy dissipation to the
+    surrounding environment at some particular rate, with some
+    given probability at a non zero temperature.
+    """
+
+    def __init__(self, p, gamma) -> None:
+        """The generalized amplitude damping channel.
+
+        Construct a channel to model the energy dissipation to the
+        environment at rate gamma, occuring with probability p
+
+        This channel evolves a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+                  + M_2 \rho M_2^\dagger + M_3 \rho M_3^\dagger
+
+        With
+            M_0 = \sqrt{p} \begin{bmatrix}
+                                1 & 0  \\
+                                0 & \sqrt{1 - \gamma}
+                           \end{bmatrix}
+            M_1 = \sqrt{p} \begin{bmatrix}
+                                0 & \sqrt{\gamma} \\
+                                0 & 0
+                           \end{bmatrix}
+            M_2 = \sqrt{1-p} \begin{bmatrix}
+                                \sqrt{1-\gamma} & 0 \\
+                                 0 & 1
+                              \end{bmatrix}
+            M_3 = \sqrt{1-p} \begin{bmatrix}
+                                 0 & 0 \\
+                                 \sqrt{gamma} & 0
+                             \end{bmatrix}
+
+        Args:
+            gamma: The damping rate
```

Comment:
I would call
p: the probability of the interaction being dissipative
gamma: the probability of the qubit and environment exchanging energy

Reply:
fixed